packages/node/src/methods/queued-execution.ts

```diff
@@ -8,18 +14,93 @@ import Queue, { Task } from "p-queue";
  * @param {Task<any>} task - any function which returns a promise-like
  * @returns
  */
-export async function addToManyQueues(queues: Queue[], task: Task<any>) {
+export async function addToManyQueues(
+  protocolOrMethodName: Protocol | Node.RpcMethodName,
+  xpub: string,
+  queueNames: string[],
+  queues: Queue[],
+  task: Task<any>,
+  lockService: Node.ILockService
+) {
   if (queues.length === 0) return await task();
 
   let promise: PromiseLike<any>;
 
+  const queuesToUnlock: string[] = [];
   /**
    * This promise will get run `n` times for `n` queues (since it
    * will be called in every queue) and so to ensure it only runs
    * once overall we memoize it.
    */
-  function runTaskWithMemoization() {
-    if (!promise) promise = task();
+  async function runTaskWithMemoization() {
+    let error;
+    let blockingQueue;
+    for (let i = 0; i < 5; i += 1) {
```

Comment:
5?

Reply:
I need a walkthru of this file when possible 🙏